packages/core/src/view/style/Stylesheet.ts

```diff
@@ -178,60 +175,36 @@ export class Stylesheet {
    * @param name Name for the style to be stored.
    * @param style Key, value pairs that define the style.
    */
-  putCellStyle(
-    name: keyof CellStateStyles,
-    style: CellStateStyles[keyof CellStateStyles]
-  ) {
-    (this.styles[name] as any) = style;
+  putCellStyle(name: string, style: CellStateStyle) {
+    this.styles.set(name, style);
   }
 
   /**
-   * Returns the cell style for the specified stylename or the given
-   * defaultStyle if no style can be found for the given stylename.
+   * Returns the cell style for the specified baseStyleName or the given
+   * defaultStyle if no style can be found for the given baseStyleName.
    *
-   * @param name String of the form [(stylename|key=value);] that represents the style.
+   * @param cellStyle An object that represents the style.
    * @param defaultStyle Default style to be returned if no style can be found.
    */
-  getCellStyle(name: string, defaultStyle: CellStateStyles) {
-    let style = defaultStyle;
-
-    if (name.length > 0) {
-      const pairs = name.split(';');
-
-      if (style && name.charAt(0) !== ';') {
-        style = clone(style);
-      } else {
-        style = {} as CellStateStyles;
-      }
-
-      // Parses each key, value pair into the existing style
-      for (const tmp of pairs) {
-        const pos = tmp.indexOf('=');
+  getCellStyle(cellStyle: CellStyle, defaultStyle: CellStateStyle) {
+    let style;
+
+    if (cellStyle.baseStyleName) {
+      // creates style with the given baseStyleName.
+      style = { ...this.styles.get(cellStyle.baseStyleName) };
+    } else if (cellStyle.baseStyleName === null) {
+      // baseStyleName is explicitly null, so don't use any default styles.
+      style = {};
+    } else {
+      style = { ...defaultStyle };
+    }
 
-        if (pos >= 0) {
-          const key = tmp.substring(0, pos) as keyof CellStateStyles;
-          const value = tmp.substring(pos + 1);
+    // Merges cellStyle into style
+    style = {
+      ...style,
+      ...cellStyle,
+    };
 
-          if (value === NONE) {
-            delete style[key];
-          } else if (isNumeric(value)) {
-            (style[key] as any) = parseFloat(value);
-          } else {
-            (style[key] as any) = value;
-          }
-        } else {
```

Comment:
We are loosing something in the new implementation.

Previously, it was possible to provide several style names according to this implementation. It wasn't explicit in the function documentation. This allowed merging various styles.
By convention, the style names were generally put at the beginning of the string, for instance `style1;style2;style3;key1=value1;key2=value2;...` but this was not mandatory (the extra style names can be put anywhere in the string).
Clearly, we miss unit tests here that would have exhibit the existing behavior 👿 .

This is something [I rely on today with mxgraph](https://github.com/process-analytics/bpmn-visualization-js/blob/v0.23.1/test/unit/component/mxgraph/renderer/StyleComputer.test.ts#L177) and that will disappear with the new implementation.
For the record, here is how it was implemented in mxgraph@4.2.2: https://github.com/jgraph/mxgraph/blob/v4.2.2/javascript/src/js/view/mxStylesheet.js#L251.

Could we change the baseStyleName to make it a string or an array of strings and update the implementation of the `getCellStyle` to support it?

Reply:
Oh, I didn't know it was possible to have multiple style names.
Making it an array of strings and having a left-to-right merge rule seems good.
I'll work on it as soon as possible.
Thanks!

Reply:
So, I renamed`baseStyleName` to `baseStyleNames` and changed it to an array of strings.
Accordingly, passing `[]` insteand of `null` clears the default styles.
Hope it works now!